woocommerce/woocommerce-ios
# Paginated list selector changes for editing linked products for a grouped product

Prerequisite for editing linked products for a grouped product in #2199 

## Changes

- Removed `PaginatedListSelectorViewController.xib` and added a table view manually in `PaginatedListSelectorViewController` so that the table view style (`.grouped`/`.plain`) can be configured in `PaginatedListSelectorViewProperties`. A table view's `tableViewStyle` cannot be changed after initialization from the xib. In #2199, I'm planning to use `PaginatedListSelectorViewController` as a child view controller with `.plain` table view style for the "Add Products" screen that is similar to the Products tab
- Added `separatorStyle: UITableViewCell.SeparatorStyle` to `PaginatedListSelectorViewProperties` since product cell used in #2199 does not use the built-in cell separator
- Set the table view style to `.grouped` for existing usage: editing product shipping classes and tax classes
- In `PaginatedListSelectorViewController`, updated UI setup based on the table view properties in `PaginatedListSelectorViewProperties`
- Since we are displaying a row type that doesn't have a xib (`ProductsTabProductTableViewCell`) in #2199, added a check to register the cell with class if the nib is not found

## Testing

A sanity check on the affected screens would be great!

- Go to the Products tab
- Tap on a simple physical product
- Tap on the price row, and tap on the tax class row --> the paginated tax classes should be shown as in `develop`, and selection should work as expected
- Go back to product details
- Tap on the shipping row, and tap on the shipping class row --> the paginated shipping classes should be shown as in `develop`, and selection should work as expected

## Example screenshots

No user-facing changes are expected!

\ | before | after
-- | -- | --
editing tax classes | ![Simulator Screen Shot - iPhone 11 - 2020-07-02 at 10 53 46](https://user-images.githubusercontent.com/1945542/86311251-87ce3600-bc52-11ea-814f-08c3563367fc.png) | ![Simulator Screen Shot - iPhone 11 - 2020-07-02 at 10 46 37](https://user-images.githubusercontent.com/1945542/86311245-83a21880-bc52-11ea-8b3d-86ea2b511480.png)
editing shipping classes | ![Simulator Screen Shot - iPhone 11 - 2020-07-02 at 10 53 56](https://user-images.githubusercontent.com/1945542/86311253-87ce3600-bc52-11ea-9cc4-0f495f6524cb.png) | ![Simulator Screen Shot - iPhone 11 - 2020-07-02 at 10 46 47](https://user-images.githubusercontent.com/1945542/86311249-869d0900-bc52-11ea-90aa-6a8af50b597f.png)

Update release notes:
